tools/cldr/cldr-to-icu/src/test/java/org/unicode/icu/tool/cldrtoicu/AlternateLocaleDataTest.java

```diff
@@ -0,0 +1,152 @@
+    }
+
+    @Test
+    public void testBadPaths() {
+        // Target and source values.
+        CldrPath target = CldrPath.parseDistinguishingPath(
+            "//ldml/numbers/currencies/currency[@type=\"USD\"]/displayName");
+        CldrPath source = CldrPath.parseDistinguishingPath(
+            "//ldml/numbers/currencies/currency[@type=\"USD\"]/symbol");
+
+        FakeDataSupplier src = new FakeDataSupplier();
+        IllegalArgumentException e = assertThrows(
+            IllegalArgumentException.class,
+            () -> AlternateLocaleData.transform(src, ImmutableMap.of(target, source)));
+        assertThat(e).hasMessageThat().contains("alternate paths must have the same namespace");
+        assertThat(e).hasMessageThat().contains(target.toString());
+        assertThat(e).hasMessageThat().contains(source.toString());
+    }
+
+    @Test
+    public void testNonLdml() {
+        // Real supplemental data with "values" in the value attributes:
+        // target: territories=[AG AR AS AU ...]
+        // source: territories=[GB]
+        // where GB is also listed as having "mon" as the first day in it's primary path.
+        //
+        // You can see why swapping paths based on 'alt' for supplemental data would be very wrong,
+        // because it would remove "XX" and "YY" by replacing the value attribute. Supplemental
+        // and BCP-47 data doesn't have a single value per path, so isn't suitable for swapping.
+        //
+        // The right way to do this would be to merge the 'territories' attribute and remove the
+        // alt territoy from its original list, but that's very complex and depends on the specific
+        // meaning of each path in question, and will probably never be supported.
+        CldrPath target = CldrPath.parseDistinguishingPath(
+            "//supplementalData/weekData/firstDay[@day=\"sun\"]");
+        CldrPath source = CldrPath.parseDistinguishingPath(
+            "//supplementalData/weekData/firstDay[@day=\"sun\"][@alt=\"variant\"]");
+
+        FakeDataSupplier src = new FakeDataSupplier();
+        IllegalArgumentException e = assertThrows(
+            IllegalArgumentException.class,
+            () -> AlternateLocaleData.transform(src, ImmutableMap.of(target, source)));
+        assertThat(e).hasMessageThat().contains("only locale data (LDML) is supported");
+        // At least one of the paths should be in the error message, so look for common substring.
+        assertThat(e).hasMessageThat().contains("/weekData/firstDay[@day=\"sun\"]");
+    }
+
+    public static void assertValuesUnordered(CldrData data, CldrValue... values) {
+        Set<CldrValue> captured = new HashSet<>();
+        data.accept(ARBITRARY, captured::add);
+        assertThat(captured).containsExactlyElementsIn(values);
+    }
+
+    private static CldrValue ldml(String path, String value) {
+        return CldrValue.parseValue("//ldml/" + path, value);
+    }
+}
```

Comment:
Damn, we can either leave these missing EOF newlines (2 or 3 that I can see, only in tests) or do a quick "in github edit" to add them. I thought IntelliJ was configured to never do this, but obviously not.

Reply:
Done in latest push